### PR TITLE
feature: Use new ingress

### PR DESCRIPTION
**Pull request by rtfpessoa**

### CODE REVIEW

File: codacy/requirements.lock [hunk 1]
```diff
@@ -11,6 +11,9 @@ dependencies:
 - name: log-router
   repository: https://charts.codacy.com/external
   version: 0.3.0
+- name: codacy-ingress
```
Comment (rtfpessoa) [inline codacy/requirements.lock]: Change to prod

File: codacy/values-production.yaml [hunk 2]
```diff
@@ -48,9 +48,29 @@ global:
...
       port: 5432
 
-codacy-api:
+codacy-ingress:
+  create: true
   ingress:
-    enabled: true ## Enable ingress
+    hostname:
+      app: <codacy-app.dns.internal> # CHANGEME: Codacy application DNS 
+      api: <codacy-api.dns.internal> # CHANGEME: Codacy api DNS
```
Comment (pauloribeiro-codacy) [inline codacy/values-production.yaml]:
```suggestion
      api: <codacy-api.dns.internal> # CHANGEME: Codacy API DNS
```

File: codacy/values-production.yaml [hunk 3]
```diff
@@ -48,9 +48,29 @@ global:
...
+    annotations:
+      kubernetes.io/ingress.class: nginx
+      kubernetes.io/tls-acme: "true"
+      certmanager.k8s.io/acme-challenge-type: http01
+      certmanager.k8s.io/cluster-issuer: letsencrypt
+      external-dns.alpha.kubernetes.io/hostname: '<codacy-app.dns.internal>, <codacy-api.dns.internal>' # CHANGEME: Codacy application and api DNS
```
Comment (pauloribeiro-codacy) [inline codacy/values-production.yaml]:
```suggestion
      external-dns.alpha.kubernetes.io/hostname: '<codacy-app.dns.internal>, <codacy-api.dns.internal>' # CHANGEME: Codacy application and API DNS 
```

File: codacy/values-production.yaml [hunk 4]
```diff
@@ -48,9 +48,29 @@ global:
...
+      certmanager.k8s.io/cluster-issuer: letsencrypt
+      external-dns.alpha.kubernetes.io/hostname: '<codacy-app.dns.internal>, <codacy-api.dns.internal>' # CHANGEME: Codacy application and api DNS 
+      external-dns.alpha.kubernetes.io/scope: public
+      nginx.ingress.kubernetes.io/proxy-body-size: 128m
+      nginx.ingress.kubernetes.io/ssl-redirect: "true"
+      nginx.ingress.kubernetes.io/use-regex: "true"
+    tls:
+      hosts:
+        - host: <codacy-app.dns.internal> # CHANGEME: Codacy application DNS 
+        - host: <codacy-api.dns.internal> # CHANGEME: Codacy api DNS
```
Comment (pauloribeiro-codacy) [inline codacy/values-production.yaml]:
```suggestion
        - host: <codacy-api.dns.internal> # CHANGEME: Codacy API DNS 
```

File: codacy/values.yaml [hunk 5]
```diff
@@ -244,6 +244,28 @@ engine:
     grafana_dashboards:
       enabled: false
 
+codacy-ingress:
+  create: false
+  ingress:
+    hostname: {}
+      # app: <codacy-app.dns.internal> # CHANGEME: Codacy application DNS 
+      # api: <codacy-api.dns.internal> # CHANGEME: Codacy api DNS
```
Comment (pauloribeiro-codacy) [inline codacy/values.yaml]:
```suggestion
      # api: <codacy-api.dns.internal> # CHANGEME: Codacy API DNS
```

File: docs/install.md [hunk 6]
```diff
@@ -278,7 +278,7 @@ codacy-postgres      1/1    12d
     -   Enable persistence on `listener`
     -   Enable persistence on `minio`
     -   Setup resources and limits
-    -   Enable the Ingress on `codacy-api`
+    -   Configure the Ingress on `values-production.yaml`
```
Comment (pauloribeiro-codacy) [inline docs/install.md]: @rtfpessoa, could you please provide just a bit more detail?

Maybe something along the lines of "follow the instructions in the comments for the `codacy-ingress` section in the `values-production.yaml` file" would be enough?

File: codacy/values.yaml [hunk 7]
```diff
@@ -244,6 +244,28 @@ engine:
     grafana_dashboards:
       enabled: false
 
+codacy-ingress:
```
Comment (pauloribeiro-codacy) [inline codacy/values.yaml]: Do we need to explicitly include this section in the values.yaml file, with "create: false"? In the documentation we are recommending that customers append the values-production.yaml file to this one, so the codacy-ingress section will be repeated and will probably trigger some conflict.

Comment (paulopontesm) [reply codacy/values.yaml]: yes. it needs to be here to pass the linting. The values.yaml usually has defaults

**Comment by joaocosta-codacy** [comment]
lgtm

**Pull request merged**